src/panels/lovelace/editor/structs/entities-struct.ts

```diff
@@ -14,6 +14,8 @@ export const entitiesConfigStruct = union([
     tap_action: optional(actionConfigStruct),
     hold_action: optional(actionConfigStruct),
     double_tap_action: optional(actionConfigStruct),
+    show_icon: optional(boolean()),
```

Comment:
That looks weird to me, since on that level according to the docs we do not support that. I think it would need to go into `buttonsHeaderFooterConfigStruct` but need to test that myself first.

Reply:
NVM, I overlooked the somewhat confusing inclusion tree. I will nonetheless create an additional PR that includes dynamic typing for the header/footer.

I also added documentation for the two fields added here: https://github.com/home-assistant/home-assistant.io/pull/22894